src/calendar/CalendarCell.ts

```diff
@@ -0,0 +1,89 @@
+import { WidgetBase } from '@dojo/widget-core/WidgetBase';
+import { ThemeableMixin, ThemeableProperties, theme } from '@dojo/widget-core/mixins/Themeable';
+import { v } from '@dojo/widget-core/d';
+import { DNode } from '@dojo/widget-core/interfaces';
+import * as css from './styles/calendar.m.css';
+
+/**
+ * @type CalendarCellProperties
+ *
+ * Properties that can be set on a Calendar Date Cell
+ *
+ * @property callFocus        Used to immediately call focus on the cell
+ * @property date             Integer date value
+ * @property disabled         Boolean, whether or not the date is in the current month
+ * @property focusable        Boolean, whether or not the date can receive focus
+ * @property selected         True if the date is currently selected
+ * @property onClick          Callback function for the click event
+ * @property onFocusCalled    Callback function when the cell receives focus
+ * @property onKeyDown        Callback function for the key down event
+ */
+export interface CalendarCellProperties extends ThemeableProperties {
+	callFocus?: boolean;
+	date: number;
+	disabled?: boolean;
+	focusable?: boolean;
+	selected?: boolean;
+	onClick?(date: number, disabled: boolean): void;
+	onFocusCalled?(): void;
+	onKeyDown?(event: KeyboardEvent): void;
+};
+
+export const CalendarCellBase = ThemeableMixin(WidgetBase);
+
+@theme(css)
+export default class CalendarCell extends CalendarCellBase<CalendarCellProperties> {
+	protected onElementCreated(element: HTMLElement, key: string) {
+		const { callFocus, onFocusCalled } = this.properties;
+		if (callFocus) {
+			element.focus();
+			onFocusCalled && onFocusCalled();
+		}
+	}
+	protected onElementUpdated(element: HTMLElement, key: string) {
+		const { callFocus, onFocusCalled } = this.properties;
```

Comment:
You could pull out the common focus logic of `onElementCreated` and `onElementUpdated` into a private method, like `_focus`.